### PR TITLE
[epoch manager] Task Metrics on EpochManager

### PR DESCRIPTION
**Pull request by Markuze**
### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3801)

### CODE REVIEW

File: consensus/src/consensus_provider.rs [hunk 1]
```diff
@@ -80,7 +80,23 @@ pub fn start_consensus(
     let (network_task, network_receiver) = NetworkTask::new(network_events, self_receiver);
 
     runtime.spawn(network_task.start());
-    runtime.spawn(epoch_mgr.start(timeout_receiver, network_receiver));
+    let metrics_monitor = tokio_metrics::TaskMonitor::new();
+    let epoch_manger_instrumneted =
+        metrics_monitor.instrument(epoch_mgr.start(timeout_receiver, network_receiver));
+
+    let metrics_frequency = std::time::Duration::from_secs(1);
```
Comment (sitalkedia) [inline consensus/src/consensus_provider.rs:87]: This is too frequent and will spam the logs, lets make it 30 s or so?

File: consensus/src/epoch_manager.rs [hunk 2]
```diff
@@ -827,28 +829,80 @@ impl EpochManager {
 
         // initial start of the processor
         self.await_reconfig_notification().await;
+
+        let consensus_messages_monitor = TaskMonitor::new();
+        let block_retrieval_monitor = TaskMonitor::new();
+        let round_timeout_monitor = TaskMonitor::new();
+
+        let metrics_frequency = std::time::Duration::from_secs(1);
```
Comment (sitalkedia) [inline consensus/src/epoch_manager.rs:837]: Ditto

File: consensus/src/consensus_provider.rs (hunk 1 above)
Comment (Markuze) [reply consensus/src/consensus_provider.rs:87]: Ack

**Review by igor-aptos — APPROVED** [review]
looks good!

**Comment by rustielin** [comment]
The Forge test looks good. Think we just forgot to cherry-pick the change that retries / (or removes) the prometheus system metrics check

**Pull request closed without merging**